examples/jsm/objects/BatchedMesh.js

```diff
@@ -490,56 +485,48 @@ class BatchedMesh extends Mesh {
 		// @TODO: Map geometryId to index of the arrays because
 		//        optimize() can make geometryId mismatch the index
 
-		const visible = this._visible;
 		const active = this._active;
 		const matricesTexture = this._matricesTexture;
-		const matrices = this._matrices;
 		const matricesArray = this._matricesTexture.image.data;
-		if ( geometryId >= matrices.length || active[ geometryId ] === false ) {
+		const geometryCount = this._geometryCount;
+		if ( geometryId >= geometryCount || active[ geometryId ] === false ) {
 
 			return this;
 
 		}
 
-		if ( visible[ geometryId ] === true ) {
-
-			matrix.toArray( matricesArray, geometryId * 16 );
-			matricesTexture.needsUpdate = true;
-
-		}
-
-		matrices[ geometryId ].copy( matrix );
+		matrix.toArray( matricesArray, geometryId * 16 );
+		matricesTexture.needsUpdate = true;
 
 		return this;
 
 	}
 
 	getMatrixAt( geometryId, matrix ) {
 
-		const matrices = this._matrices;
 		const active = this._active;
-		if ( geometryId >= matrices.length || active[ geometryId ] === false ) {
+		const matricesArray = this._matricesTexture.image.data;
+		const geometryCount = this._geometryCount;
+		if ( geometryId >= geometryCount || active[ geometryId ] === false ) {
 
-			return matrix;
+			return null;
```

Comment:
If there is no data for the matrix then we just return null here to indicate there's no relevant value.